src/service/resource.js

```diff
@@ -895,6 +895,14 @@ export class Resource {
     );
     devAssert(this.isDisplayed(), 'Not displayed for layout: %s', this.debugid);
 
+    if (this.state_ != ResourceState.LAYOUT_SCHEDULED) {
```

Comment:
@jridgewell not sure if this gets into the codepath of forced layouts. For instance, `amp-animation` forces the layout on the elements used in the animation, I believe.

Reply:
I imagine you're talking about `requireLayout`? If so, it already directly checks for `LAYOUT_SCHEDULED` (https://github.com/ampproject/amphtml/blob/156ab5916cf04f1ed6bc0e41905c1d791a0a7704/src/service/owners-impl.js#L119-L134). It also goes through `scheduleLayoutOrPreload` (which is the same thing regular resource scheduling uses), and will set the resource to `LAYOUT_SCHEDULED`.

Reply:
Got it. Yes. I think `requireLayout` covers this.